mathesar_ui/src/sections/table-view/constraints/TableConstraint.svelte

```diff
@@ -0,0 +1,103 @@
+<script lang='ts'>
+  import { fly } from 'svelte/transition';
+  import type { FlyParams } from 'svelte/transition';
+  import {
+    Icon,
+    Seesaw,
+    Button,
+    Spinner,
+  } from '@mathesar-components';
+  import type { Constraint } from '@mathesar/stores/table-data/constraints';
+  import {
+    faTrash,
+    faCheck,
+    faArrowLeft,
+    faExclamationTriangle,
+  } from '@fortawesome/free-solid-svg-icons';
+  
+  export let constraint: Constraint;
+  export let drop: () => Promise<void>;
+
+  let isConfirmingDrop = false;
+  let isSubmittingDrop = false;
+  let useTransitionOut = false;
+
+  async function handleDrop() {
+    isSubmittingDrop = true;
+    useTransitionOut = true;
+    try {
+      await drop();
+    } catch (error) {
+      // TODO display toast message with error
+      useTransitionOut = false;
+    } finally {
+      isSubmittingDrop = false;
+    }
+  }
+  
+  $: columnSummary = constraint.columns.join(', ');
+  $: transitionDuration = useTransitionOut ? 200 : 0 as number;
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  $: transition = { x: 200, duration: transitionDuration } as FlyParams;
+</script>
+
+<div class='table-constraint' class:isConfirmingDrop class:isSubmittingDrop>
```

Comment:
It would be better if we followed consistent css class naming pattern, in our case we've followed kebab-cased css class naming everywhere.

```suggestion
<div class='table-constraint' class:is-confirming-drop={isConfirmingDrop} class:is-submitting-drop={isSubmittingDrop}>
```

Reply:
- Yeah this is that code I mentioned I had already written when we chatted on the phone about this.
- Turns out I wasn't using the `isConfirmingDrop` class. So I removed it.
- I fixed the `isSubmittingDrop` class in af011153ce1ed7b90025307728c08da8c9865796